src/routeGeneration/templateHelpers.ts

```diff
@@ -362,6 +366,130 @@ export class ValidationService {
+  ): any {
+    if (!subSchemas) { return; }
+
+    const subFieldErrors: FieldErrors[] = [];
+    const values: any[] = [];
+
+    subSchemas.forEach(subSchema => {
+      const subFieldError: FieldErrors = {};
+      const val = this.ValidateParam(subSchema, JSON.parse(JSON.stringify(value)), name, subFieldError, parent, minimalSwaggerConfig);
+      subFieldErrors.push(subFieldError);
+      values.push(val);
+    });
+
+    if (subFieldErrors.length > 0 && !subFieldErrors.some(subFieldError => Object.keys(subFieldError).length === 0)) {
+      fieldErrors[parent + name] = {
+        message: `Could not match the union against any of the items. Issues: ${JSON.stringify(subFieldErrors)}`,
+        value,
+      };
+      return;
+    }
+
+    if (value instanceof Object && minimalSwaggerConfig.noImplicitAdditionalProperties === 'silently-remove-extras') {
+      return values.reduce((acc, obj) => ({...acc, ...obj}), {});
+    }
+    return value;
+  }
+
+  public validateIntersection(
+    name: string,
+    value: any,
+    fieldErrors: FieldErrors,
+    swaggerConfig: SwaggerConfigRelatedToRoutes,
+    subSchemas: TsoaRoute.PropertySchema[] | undefined,
+    parent = '',
+  ): any {
+    if (!subSchemas) { return; }
+
+    const subFieldErrors =
+      subSchemas
+        .filter(subSchema => subSchema.ref)
+        .map(subSchema => {
+          const subFieldErrors: FieldErrors = {};
+          this.validateModel({
+            fieldErrors: subFieldErrors,
+            minimalSwaggerConfig: { noImplicitAdditionalProperties: undefined },
+            name,
+            parent,
+            refName: subSchema.ref as string,
+            value,
+          });
+          return subFieldErrors;
+        });
+
+    const filtered = subFieldErrors.filter(subFieldError => Object.keys(subFieldError).length !== 0);
+
+    if (filtered.length > 0) {
+      fieldErrors[parent + name] = {
+        message: `Could not match the intersection against every type. Issues: ${JSON.stringify(filtered)}`,
+        value,
+      };
+      return;
+    }
+
+    // Only Model definitions make sense here right now
+    const refNames = subSchemas.filter(subschema => subschema.ref).map(subschema => subschema.ref) as string[];
+    if (!refNames.every(refName => this.models[refName])) {
+      return value;
+    }
+
+    const reportedExcess = new Set(refNames.map(refName => this.models[refName]).reduce((acc, subSchema) => {
+      return [...acc, ...this.getExcessPropertiesFor(subSchema, Object.keys(value), swaggerConfig)];
+    }, []));
+
+    if (reportedExcess.size === 0) {
+      return value;
+    }
+
+    const allowedProperties = new Set(refNames.map(refName => this.models[refName]).reduce((acc, subSchema) => {
+      return [...acc, ...this.getPropertiesFor(subSchema)];
+    }, []));
+
+    const actualExcess = [...reportedExcess].filter(property => !allowedProperties.has(property));
+
+    if (actualExcess.length > 0) {
+      fieldErrors[parent + name] = {
```

Comment:
@WoH we should only set these field errors if the tsoa user has configured it to throw. So I believe that this should be replaced with:
```ts
    if (actualExcess.length > 0) {
      if (swaggerConfig.noImplicitAdditionalProperties === 'throw-on-extras' || swaggerConfig.noImplicitAdditionalProperties === true){
        fieldErrors[parent + name] = {
          message: `The following properties are not allowed by any part of the intersection: ${actualExcess.join(', ')}`,
          value,
        };
      } else if (swaggerConfig.noImplicitAdditionalProperties === 'silently-remove-extras'){
        actualExcess.forEach(excessKey => {
          delete value[excessKey];
        });
      } else if (swaggerConfig.noImplicitAdditionalProperties === false || !swaggerConfig.noImplicitAdditionalProperties){
        // then excess properties are acceptable
      } else {
        assertNever(swaggerConfig.noImplicitAdditionalProperties)
      }
    }
```

Reply:
If there is reported excess, that means the user has already set it to throw.
See [here](https://github.com/lukeautry/tsoa/pull/400/files#diff-054445345683aa77f57a22be4b3d8b28R473), which is the helper that determines reported Excess for one object/model.
If the user sets 'remove', the value return from model checking will be stripped from additional props -> nothing else to do
If no setting -> reported excess is [] -> actual excess is [].

Reply:
I don’t understand your response because the unit tests that I provided for you don’t pass unless that code is in there.

Reply:
The setup of one specific test was incorrect. It should've been passing `{ value1: '' }`, not `{ or: {value1: '' }}`.
On the silently removing stuff, yes, union did that already, intersection now does the same logical setup. Therefore the removing excess works aswell.

Reply:
Also, please note: You verified vs. the dataToValidate, not the result in some cases where that really matters. The result is correct.